src/FormField/RegisteredMFAMethodListField.php

```diff
@@ -2,17 +2,43 @@
 
 namespace SilverStripe\MFA\FormField;
 
+use InvalidArgumentException;
 use SilverStripe\Admin\SecurityAdmin;
 use SilverStripe\Forms\FormField;
 use SilverStripe\MFA\Controller\AdminRegistrationController;
 use SilverStripe\MFA\Model\RegisteredMethod;
 use SilverStripe\MFA\Service\MethodRegistry;
 use SilverStripe\MFA\Service\RegisteredMethodManager;
 use SilverStripe\MFA\Service\SchemaGenerator;
+use SilverStripe\Security\Member;
 use SilverStripe\Security\Security;
 
 class RegisteredMFAMethodListField extends FormField
 {
+    public function __construct($name, $title, $value)
+    {
+        // Because {@see setValue()} can do nothing on e.g. form submission, it is of critical importance that this
+        // object is never in existence without a valid Member set as the value - we have set the `$value` parameter as
+        // required, however we still need to ensure that it is a valid value.
+        if (!$value instanceof Member) {
+            throw new InvalidArgumentException(__CLASS__ . ' requires a Member to be set at all times');
+        }
```

Comment:
My suggestion would be to move this to constructor arg type hints, then you'll get a runtime exception instead:

```php
public function __construct(string $name, string $title, Member $value)
```

Reply:
And that's still legit?
Golly!